### PR TITLE
Don't retry from retry

### PR DESCRIPTION
**Pull request by ekr**
This was split off from PR#609. Hopefully this is uncontroversial.

### CODE REVIEW

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -885,9 +885,9 @@ version was negotiated, the client can regard ECH as securely disabled
 by the server, and it SHOULD retry the handshake with a new transport
 connection and ECH disabled.
 
-Clients SHOULD implement a limit on retries caused by receipt of "retry_configs"
-or servers which do not acknowledge the "encrypted_client_hello" extension. If
-the client does not retry in either scenario, it MUST report an error to the
+Clients SHOULD NOT accept "retry_config" in response to
+a connection initiated in response to a "retry_config".
```
Comment (martinthomson) [inline draft-ietf-tls-esni.md:889]: Do we have a little more color to add here?

This is mostly about the case where a retry config is supplied by one server instance, but other server instances don't work.  That is a server configuration error, unless the client decided to keep the retry config for too long, in which case it is a signal that the client shouldn't hold on to old state.

Adding that sort of context might make the SHOULD a little more useful.

**Comment by ekr** [comment]
I added some text.

File: draft-ietf-tls-esni.md [hunk 2]
```diff
@@ -885,10 +885,13 @@ version was negotiated, the client can regard ECH as securely disabled
...
-Clients SHOULD implement a limit on retries caused by receipt of "retry_configs"
-or servers which do not acknowledge the "encrypted_client_hello" extension. If
-the client does not retry in either scenario, it MUST report an error to the
-calling application.
+Clients SHOULD NOT accept "retry_config" in response to a connection
+initiated in response to a "retry_config".  Sending a "retry_config"
+in this situation is a signal that the server is misconfigured, e.g.,
+the server might have multiple inconsistent configurations so that the
+client reached a node with configuration A in the first connection and
+a node with configuration B in the second. If a client does not retry,
```
Comment (elardus-sophos) [inline draft-ietf-tls-esni.md:893]: The previous paragraph also talks about a "retry". But that is an ECH *disabled* retry.
In this paragraph, based on the context, it is referring to an ECH *enabled* retry.
Should the client be allowed to perform an ECH disabled retry here? And should anything about that, allowed or not allowed, be noted in this paragraph also?

Comment (ekr) [reply draft-ietf-tls-esni.md]: I think this is a reasonable question. Updated the text.

Note that I also broke out the requirement to report the error separately as I think it applies to both cases.

**Comment by ekr** [comment]
@martinthomson?

**Review by martinthomson — APPROVED** [review]
What is the merge conflict?

**Comment by ekr** [comment]
I had two retry change PRs off the same place and the first one merged.

**Pull request merged**